docs/notary.md

```diff
@@ -0,0 +1,531 @@
+`NotValidBefore`, `Conflicts` and `NotaryAssisted` transaction attribute, handle,
+verify and broadcast `P2PNotaryRequest` P2P payloads, properly initialize native
+Notary contract and designate `P2PNotary` node role in RoleManagement native
+contract.
+
+If you use custom `NativeActivations` subsection of the `ProtocolConfiguration`
+section in your node config, then specify the height of the Notary contract
+activation, e.g. `0`.
+
+Note, that even if `P2PSigExtensions` config subsection enables notary-related
+logic in the network, it still does not turn your node into notary service node.
+To enable notary service node functionality refer to the
+[NeoGo Notary service](#NeoGo-Notary-service-node-module) documentation.
+
+### Example
+
+```
+  P2PSigExtensions: true
+  NativeActivations:
+    Notary: [0]
+    ContractManagement: [0]
+    StdLib: [0]
+    CryptoLib: [0]
+    LedgerContract: [0]
+    NeoToken: [0]
+    GasToken: [0]
+    PolicyContract: [0]
+    RoleManagement: [0]
+    OracleContract: [0]
+```
+
+
+## NeoGo Notary service node module
+
+NeoGo node can act as notary service node (the node that accumulates notary
+requests, collects signatures and releases fully-signed transactions). It has to
+have a wallet with key belonging to one of network's designated notary nodes
+(stored in `RoleManagement` native contract). Also, the node must be connected to
+the network with enabled P2P signature extensions, otherwise problems with states
+and peer disconnections will occur.
+
+Notary service node doesn't need [RPC service](rpc.md) to be enabled, because it
+receives notary requests and broadcasts completed transactions via P2P protocol.
+However, enabling [RPC service](rpc.md) allows to send notary requests directly
+to the notary service node and avoid P2P communication delays.
+
+### Configuration
+
+To enable notary service node check firstly that
+[P2PSignatureExtensions](#NeoGo P2P signature extensions) are properly set up.
+Then add `P2PNotary` subsection to `ApplicationConfiguration` section of your
+node config.
+
+Parameters:
+* `Enabled`: boolean value, enables/disables the service node, `true` for service
+  node to be enabled
+* `UnlockWallet`: notary node wallet configuration:
+    - `Path`: path to NEP-6 wallet.
+    - `Password`: password for the account to be used by notary node.
+
+### Example
+
+```
+P2PNotary:
+  Enabled: true
+  UnlockWallet:
+    Path: "/notary_node_wallet.json"
+    Password: "pass"
+```
+
+
+## Notary request lifecycle guide
+
+Below presented stages each P2P signature collection request goes through. Use
+stages 1 and 2 to create, sign and submit P2P notary request. Stage 3 is
+performed by the notary service, do not require user's intervention and is given
+for informational purposes. Stage 4 contains tips to check notary request
+results.
+
+### 1. Notary deposit
+
+Notary service is not free and is paid through the sender's deposit that
```

Comment:
This is not entirely correct, normally it's paid for by transactions, deposit just guarantees that payment will still be done if things go wrong.